core/src/main/java/org/apache/accumulo/core/volume/VolumeConfiguration.java

```diff
@@ -107,29 +110,31 @@ public static String getConfiguredBaseDir(AccumuloConfiguration conf,
 
         try {
           // pass through URI to unescape hex encoded chars (e.g. convert %2C to "," char)
-          configuredBaseDirs[i++] = new Path(new URI(namespace)).toString();
+          configuredBaseDirs.add(new Path(new URI(namespace)).toString());
         } catch (URISyntaxException e) {
           throw new IllegalArgumentException(Property.INSTANCE_VOLUMES.getKey() + " contains "
               + namespace + " which has a syntax error", e);
         }
       }
     }
 
-    return configuredBaseDirs;
+    LinkedHashSet<String> deduplicated = new LinkedHashSet<>();
+    deduplicated.addAll(configuredBaseDirs);
+    if (deduplicated.isEmpty()) {
+      throw new IllegalArgumentException(
+          Property.INSTANCE_VOLUMES.getKey() + " contains no volumes (" + ns + ")");
+    }
+    if (deduplicated.size() < configuredBaseDirs.size()) {
+      throw new IllegalArgumentException(
+          Property.INSTANCE_VOLUMES.getKey() + " contains duplicate volumes (" + ns + ")");
```

Comment:
Can never have enough sanity checks!  Could create unit tests for these checks, one where two configured Volumes get normalized to the same Path.

Reply:
This code is likely to change with #1397, so I didn't want to spend too much time adding more stuff like that just yet, but can do so with #1397.